docs/presentations/articles/rasa.md

```diff
@@ -0,0 +1,132 @@
+
+_Author: [Andreas Loeffler](mailto:an161loe@htwg-konstanz.de)_
+
+## TL;DR
+
+_Rasa_ is an open-source conversational AI framework with two main components: Rasa NLU for natural language understanding and Rasa Core for dialogue management. It stands out with its contextual understanding, scalability, and open-source flexibility.
+
+## What is Rasa?
+
+[Rasa](https://rasa.com/) is an open-source conversational AI framework that enables developers to build robust and natural language understanding (NLU) models for chatbots and virtual assistants. It goes beyond simple rule-based approaches and allows for the creation of intelligent, context-aware conversational agents.
+
+## Key Components of Rasa
+
+### 1. Rasa NLU
+
+Rasa NLU is the natural language understanding component of the framework. It is responsible for extracting intents and entities from user messages. With support for both rule-based and machine learning-based approaches, Rasa NLU allows developers to train models that can understand user input in a contextual manner.
+
+```markdown
+# Example Rasa NLU Training Data
+
+## intent:greet
+
+- hey
+- hello
+- hi
+
+## intent:goodbye
+
+- bye
+- farewell
+- see you later
+
+## intent:book_flight
+
+- I want to book a flight to [New York](destination)
+- Book a flight from [London](origin) to [Paris](destination)
+```
+
+### 2. Rasa Core
+
+Rasa Core is the dialogue management component that handles the flow of the conversation. It decides how the chatbot should respond to user inputs based on the current context and the predicted intents and entities from Rasa NLU. This allows for dynamic and contextually aware conversations.
+
+```markdown
+# Example Rasa Core Dialogue
+
+- greet
+  - utter_greet
+- book_flight
+  - action_check_flight_availability
+  - utter_confirm_booking
+- goodbye
+  - utter_goodbye
+```
+
+### 3. Rasa Actions
+
+Rasa Actions are custom code snippets that define the behavior of the chatbot. They can perform actions such as calling APIs, querying databases, or any other custom logic required for the conversation. This extensibility makes Rasa suitable for a wide range of applications.
+
+```python
+# Example Rasa Action Code
+
+class ActionCheckFlightAvailability(Action):
+    def name(self) -> Text:
+        return "action_check_flight_availability"
+
+    def run(self, dispatcher, tracker, domain):
+        # Custom logic to check flight availability
+        dispatcher.utter_message("Yes, flights are available.")
+        return []
+```
+
+## Rasa in Action
+
+Let's walk through a simple scenario to illustrate how Rasa works:
+
+1. **User Input:** "I want to book a flight from New York to London."
+
+2. **Rasa NLU Output:**
+
+   - Intent: `book_flight`
+   - Entities: `origin=New York`, `destination=London`
+
+3. **Rasa Core Decision:**
+
+   - Trigger `book_flight` action
+
+4. **Rasa Action:**
+
+   - `action_check_flight_availability` is executed, checking flight availability and responding accordingly.
```

Comment:
I think the indentation comes out wrong:

![image](https://github.com/pkeilbach/htwg-practical-nlp/assets/28002199/43e800fc-57da-4c61-9924-444378953c6f)


I think you need to indent the unordered sub-points by 4 spaces. Also make sure to have 4 spaces for the ordered part:

```md
2.  **Rasa NLU Ouput** (2 spaces between . and *)
    - Intent: `book_flight` (4 spaces before the -)
    - Entities: `origin=New York`, `destination=London`
```

But you can check if it comes out as intended by bringing up mkdocs locally:

```
make lecture_notes
```